src/components/CommentsV2/commentv2.utils.ts

```diff
@@ -0,0 +1,45 @@
+import { ToggleHideCommentInput } from '~/server/schema/commentv2.schema';
+import { showErrorNotification } from '~/utils/notifications';
+import { trpc } from '~/utils/trpc';
+import { CommentConnectorInput } from '~/server/schema/commentv2.schema';
+
+export const useQueryThreadComments = (filters: CommentConnectorInput) => {
+  const { data, isLoading: loadingComments } = trpc.commentv2.getThreadDetails.useQuery(filters);
+  const { data: count = 0, isLoading: loadingCount } = trpc.commentv2.getCount.useQuery(filters);
+
+  return {
+    comments: data?.comments ?? [],
+    loadingComments,
+    count,
+    loadingCount,
+  };
+};
+
+export const useMutateComment = () => {
+  const queryUtils = trpc.useContext();
+  const toggleHideCommentMutation = trpc.commentv2.toggleHide.useMutation({
+    async onSuccess(_, { entityType, entityId }) {
+      await queryUtils.commentv2.getThreadDetails.invalidate({ entityType, entityId });
+      await queryUtils.commentv2.getThreadDetails.invalidate({
+        entityType,
+        entityId,
+        hidden: true,
+      });
+      await queryUtils.commentv2.getCount.invalidate({ entityType, entityId });
+      await queryUtils.commentv2.getCount.invalidate({ entityType, entityId, hidden: true });
```

Comment:
I'd avoid another fetch just for count, might be better to just optimistic update

Reply:
Indeed, just went with the fasted approach for the time being. Planning on adding optimistic updates in next iteration